File: 2394-count-subarrays-with-score-less-than-k/solution.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        result = 0
        sum = 0
        left = 0
        for right in range(len(nums)):
            sum += nums[right]
            while sum * (right - left + 1) >= k:
                sum -= nums[left]
                left += 1
            result += (right - left + 1)
        return result

File: 2394-count-subarrays-with-score-less-than-k/solution.rs
impl Solution {
    pub fn count_subarrays(mut nums: Vec<i32>, k: i64) -> i64 {
        let mut result: i64 = 0;
        let mut sum: i64 = 0;
        let mut left: usize = 0;
        for right in 0 ..nums.len() {
            sum += nums[right] as i64;
            while sum * (right - left + 1) as i64 >= k {
                sum -= nums[left] as i64;
                left += 1;
            }
            result += (right - left + 1) as i64;
        }
        result
    }
}
